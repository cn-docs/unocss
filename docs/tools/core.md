---
title: 核心
description: UnoCSS 的核心引擎，不包含任何预设。它可以作为你自己原子 CSS 框架的引擎使用。
---

# 核心

UnoCSS 的核心引擎（无预设）：`@unocss/core`。它可以作为你自己原子 CSS 框架的引擎使用。

## 安装

::: code-group

```bash
pnpm add -D @unocss/core
```

```bash
yarn add -D @unocss/core
```

```bash
npm install -D @unocss/core
```

:::

## 使用

```ts
import { createGenerator } from '@unocss/core'

const generator = await createGenerator(
  { /* 用户选项 */ },
  { /* 默认选项 */ }
)

const { css } = await generator.generate(code)
```

## 许可证

- MIT 许可证 &copy; 2021-PRESENT [Anthony Fu](https://github.com/antfu)
