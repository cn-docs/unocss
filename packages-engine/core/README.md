# @unocss/core

The core engine of [UnoCSS](https://github.com/unocss/unocss) without any presets. It can be used as the engine of your own atomic CSS framework.

## Documentation

Please refer to the [documentation](https://unocss.dev/tools/core).

## License

MIT License &copy; 2021-PRESENT [Anthony Fu](https://github.com/antfu)
