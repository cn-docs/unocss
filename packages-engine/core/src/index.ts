export * from './config'
export * from './constants'
export * from './extractors'
export * from './generator'
export * from './types'
export * from './utils'
