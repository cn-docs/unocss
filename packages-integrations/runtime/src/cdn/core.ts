import init from '../index'

init()
