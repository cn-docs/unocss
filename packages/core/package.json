{
  "name": "@unocss/core",
  "type": "module",
  "version": "0.62.3",
  "description": "The instant on-demand Atomic CSS engine.",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "license": "MIT",
  "funding": "https://github.com/sponsors/antfu",
  "homepage": "https://github.com/unocss/unocss/tree/main/packages/core#readme",
  "repository": {
    "type": "git",
    "url": "https://github.com/unocss/unocss",
    "directory": "packages/core"
  },
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.mts",
      "default": "./dist/index.mjs"
    }
  },
  "main": "dist/index.mjs",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "devDependencies": {
    "magic-string": "catalog:",
    "unconfig": "catalog:"
  }
}
