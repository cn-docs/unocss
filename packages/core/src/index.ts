export * from './config'
export * from './extractors'
export * from './generator'
export * from './types'
export * from './utils'
