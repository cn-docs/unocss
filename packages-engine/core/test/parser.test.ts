import { createGenerator } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import { expect, it } from 'vitest'

it('split string with custom separator', async () => {
  const uno = await createGenerator({
    presets: [
      presetUno(),
    ],
    separators: ['__'],
  })

  const { css } = await uno.generate('backdrop__shadow-green', { preflights: false })
  expect(css).toMatchSnapshot()
})

it('unable to generate token variant with explicit separator without dash', async () => {
  const uno = await createGenerator({
    presets: [
      presetUno(),
    ],
    separators: '-',
  })

  const { css } = await uno.generate('backdrop-shadow-green', { preflights: false })
  expect(css).eql('')
})
