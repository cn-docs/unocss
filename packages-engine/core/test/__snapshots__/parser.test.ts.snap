// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`split string with custom separator 1`] = `
"/* layer: default */
.backdrop__shadow-green::backdrop{--un-shadow-opacity:1;--un-shadow-color:rgb(74 222 128 / var(--un-shadow-opacity));}"
`;
